lib/node_modules/@stdlib/utils/group/docs/types/index.d.ts

```diff
@@ -51,7 +51,10 @@ interface Options {
 * var out = group( arr, groups );
 * // returns { 'b': [ 'beep', 'boop', 'bar' ], 'f': [ 'foo' ] }
 */
-declare function group( collection: Collection, groups: Collection ): any;
+declare function group<T, K extends string | number>(
+	collection: Collection<T>,
+	groups: Collection<K>
+): { [key: string]: Collection<T> };
```

Comment:
Let's use built-in `PropertyKey` instead (which also encompasses symbols, which are valid property keys):

```suggestion
declare function group<T, K extends PropertyKey>(
	collection: Collection<T>,
	groups: Collection<K>
): { [key: PropertyKey]: Collection<T> };
```